### ISSUE TITLE
Make a V-Sekai Central launcher

### ISSUE THREAD

**Issue by fire** [V-Sekai/manuals#452]
### Metadata

- Status: proposed 
- Deciders: V-Sekai,
- Tags: V-Sekai,

### Context and Problem Statement

It is cumbersome to launch V-Sekai.

### Describe the proposed option and how it helps to overcome the problem or limitation

Make a custom launcher for V-Sekai.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, or diagrams

Use the state chart plugin for ui.

```gdscript
const games = {
    "game1": "path/to/game1",
    "game2": "path/to/game2",
    "game3": "path/to/game3"
}

func browse_games():
    print("Games available on v-sekai:")
    for game in games.keys():
        print(game)

func launch_game(game: String):
    if games.has(game):
        print("Launching ", game, "...")
        OS.execute(games[game])
    else:
        print(game, " not found.")

func check_updates():
    # Code to check for game updates

func get_updates():
    # Code to download game updates

func receive_news():
    # Code to retrieve news and updates

# Main program loop
while true:
    print("Welcome to the v-sekai game launcher.")
    print("Please select an option:")
    print("1. Browse games")
    print("2. Launch game")
    print("3. Check for updates")
    print("4. Receive news and updates")
    print("5. Quit")
    var choice = Input.ask("Enter choice: ")

    match choice:
        "1":
            browse_games()
        "2":
            var game = Input.ask("Enter game name: ")
            launch_game(game)
        "3":
            check_updates()
        "4":
            receive_news()
        "5":
            print("Goodbye!")
            break
        _:
            print("Invalid choice. Please try again.")
```

### Positive Consequences

We are able to use V-Sekai easier.

### Negative Consequences

We cannot get other features in.

### Option graveyard

We can't do nothing.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It is a few lines of script.

### Is there a reason why this should be core and done by us?

We need to own the game experience.

### References

- [V-Sekai](https://v-sekai.org/)

**Comment by fire** [comment]
```
Test Protocol #2 for zstd
Download
https://downloads.tuxfamily.org/godotengine/3.2.3/rc3/Godot_v3.2.3-rc3_win64.exe.zip
https://downloads.tuxfamily.org/godotengine/3.2.3/rc1/Godot_v3.2.3-rc1_win64.exe.zip
 zstd.exe --patch-from=.\Godot_v3.2.3-rc1_win64.exe .\Godot_v3.2.3-rc3_win64.exe -o patch
17.8 MB 
Feels instant
zstd.exe --patch-from=.\Godot_v3.2.3-rc1_win64.exe .\Godot_v3.2.3-rc3_win64.exe -o patch -19 --zstd=targetLength=4096 --zstd=chainLog=30 --single-thread
8.99 MB
```

**Comment by fire** [comment]
Use sqlite from range query to download the chunks.

**Comment by fire** [comment]
https://github.com/vslavik/winsparkle is simpler

**Comment by fire** [comment]
In favour of desync https://github.com/folbricht/desync

**Comment by fire** [comment]
https://www.daemonology.net/blog/2009-06-24-encrypt-then-mac.html

1. I don't know what version the url to the latest build of v-sekai for windows x86, it will update every update.
2. secondly it needs a signature to prove who distributed it and provenance

**Comment by fire** [comment]
Use `syft attest`

**Comment by fire** [comment]
https://www.redhat.com/en/blog/cryptographic-signatures-zip-distributions

1. Take all hashes and put them in SHA256SUM
2. Generate a SHA256SUM, SHA256SUM.asc, and README.md file

**Comment by fire** [comment]
See also https://www.chosenplaintext.ca/articles/signing-files-vs-signing-hashes.html

**Comment by fire** [comment]
[mingw-w64-rhash](https://packages.msys2.org/base/mingw-w64-rhash)

**Comment by fire** [comment]
```
https://github.com/fire.keys I need a folder I can upload the split cache files and I can host the index files on github's git repos which handles the security problem and I need a url for the folder to upload
the security problem is not satisfied by hosting data on URLs
so we can borrow that tech and a url
Host the index files on github's git repos which handles the security problem because we can gpg sign both the commit and next to the index file
1. https://uro.v-sekai.cloud/storage01/store is based on content hashing
2.[ https://raw.githubusercontent.com/v-sekai/vsk-index/master/vsekai_game_linux_x86_64.caidx](https://raw.githubusercontent.com/v-sekai/vsk-index/master/vsekai_game_linux_x86_64.caidx) and[ https://raw.githubusercontent.com/v-sekai/vsk-index/master/vsekai_game_linux_x86_64.caidx.asc](https://raw.githubusercontent.com/v-sekai/vsk-index/master/vsekai_game_linux_x86_64.caidx.asc) can be used to verify the cache and index them 

https://github.com/Rich-Harris/degit

scoop install enc
 Get-Content ..\VeraCrypt_PGP_public_key.asc | .\enc.exe key info

cat encrypted.bin | enc sig verify --key public.key --signature message.sig

Upload gpg public keys to github.

enc remote get --provider=github fire

.\enc.exe remote get --provider=github fire > fire.key

cat enc.exe | .\enc.exe sig verify --key fire.key --signature enc.sig

tl;dr

mkdir vsekai_game_casync
.\desync.exe tar -i -s vsekai_game_casync/store vsekai_game_casync/vsekai_game_linux_x86_64.caidx vsekai_game_linux_x86_64
.\desync.exe tar -i -s vsekai_game_casync/store vsekai_game_casync/vsekai_game_windows_x86_64.caidx vsekai_game_windows_x86_64
desync untar -s v-sekai-game-casync/store -i vsekai_game_windows_x86_64.caidx .
scoop install vsekai_game

Scoop uses sha256 hashes
```

**Comment by fire** [comment]
Pause for now.
